Document Guild Scheduled Event structure and HTTP methods (#3586)
hammerandchisel/discord-api-docs: docs/topics/Gateway.md

| [Guild Role Update](#DOCS_TOPICS_GATEWAY/guild-role-update)                         | guild role was updated                                                                                                           |
| [Guild Role Delete](#DOCS_TOPICS_GATEWAY/guild-role-delete)                         | guild role was deleted                                                                                                           |
| [Guild Scheduled Event Create](#DOCS_TOPICS_GATEWAY/guild-scheduled-event-create)    | guild scheduled event was created                                                                                               |
| [Guild Scheduled Event Update](#DOCS_TOPICS_GATEWAY/guild-scheduled-event-update)    | guild scheduled event was updated                                                                                               |
| [Guild Scheduled Event Delete](#DOCS_TOPICS_GATEWAY/guild-scheduled-event-delete)    | guild scheduled event was deleted                                                                                               |

> | [Guild Scheduled Event User Create](#DOCS_TOPICS_GATEWAY/guild-scheduled-event-user-create) | user subscribed to a guild scheduled event                                                                               |
> | [Guild Scheduled Event User Delete](#DOCS_TOPICS_GATEWAY/guild-scheduled-event-user-delete) | user unsubscribed from a guild scheduled event                                                                           |

Reviewer: ```suggestion
```
`GUILD_SCHEDULED_EVENT_USER_CREATE` and `GUILD_SCHEDULED_EVENT_USER_DELETE` will be deprecated in the near future so we can remove these

Reviewer 2: We should replace it I guess.

| guild_id | snowflake | id of the guild |
| role_id  | snowflake | id of the role  |

### Guild Scheduled Event Create

> Sent when a guild event is created. The data contain a [Guild Event](#DOCS_RESOURCES_GUILD_EVENT/guild-event-object) object.

Reviewer 3: ```suggestion
Sent when a guild event is created. The inner payload is a [Guild Event](#DOCS_RESOURCES_GUILD_EVENT/guild-event-object) object.
```
This and the Update and Delete descriptions should be consistent with the others.

| guild_id | snowflake | id of the guild |
| role_id  | snowflake | id of the role  |

### Guild Scheduled Event Create

Sent when a guild event is created. The inner payload is a [Guild Event](#DOCS_RESOURCES_GUILD_EVENT/guild-event-object) object.

### Guild Scheduled Event Update

Sent when a guild event is updated. The inner payload is a [Guild Event](#DOCS_RESOURCES_GUILD_EVENT/guild-event-object) object.

### Guild Scheduled Event Delete

Sent when a guild event is deleted. The inner payload is a [Guild Event](#DOCS_RESOURCES_GUILD_EVENT/guild-event-object) object. See [Guild Scheduled Event Status](#DOCS_RESOURCES_GUILD_EVENT/guild-event-object-guild-scheduled-event-status) to determine how the event was deleted.

### Guild Scheduled Event User Create

Sent when a user has subscribed to a Guild Event.

###### Guild Scheduled Event User Create Event Fields

| Field                    | Type      | Description           |
| ------------------------ | --------- | --------------------- |
| guild_scheduled_event_id | snowflake | id of the guild event |

> | user_id                  | snowflake | id of the user        |

Reviewer 4: A `guild_id` field would have been really helpful in the payload of these two `GUILD_SCHEDULED_EVENT_USER_*` events

Author: @msciotti Any thoughts?

Reviewer: we've changed it to include `guild_id` now :) but the event names will be changed. I'll explain in a separate comment

> | guild_id | snowflake | id of the guild |

Reviewer 5: ```suggestion
| guild_id | snowflake | ID of the guild |
```

> | role_id  | snowflake | id of the role  |

Reviewer 5: ```suggestion
| role_id  | snowflake | ID of the role  |
```

### Guild Scheduled Event Create

Sent when a guild scheduled event is created. The inner payload is a [guild scheduled event](#DOCS_RESOURCES_GUILD_SCHEDULED_EVENT/guild-scheduled-event-object) object.

### Guild Scheduled Event Update

Sent when a guild scheduled event is updated. The inner payload is a [guild scheduled event](#DOCS_RESOURCES_GUILD_SCHEDULED_EVENT/guild-scheduled-event-object) object.

### Guild Scheduled Event Delete

Sent when a guild scheduled event is deleted. The inner payload is a [guild scheduled event](#DOCS_RESOURCES_GUILD_SCHEDULED_EVENT/guild-scheduled-event-object) object. See [Guild Scheduled Event Status](#DOCS_RESOURCES_GUILD_SCHEDULED_EVENT/guild-scheduled-event-object-guild-scheduled-event-status) to determine how the event was deleted.

### Guild Scheduled Event User Add

Sent when a user has subscribed to a guild scheduled event.

> ###### Guild Scheduled Event User Create Event Fields

Reviewer 2: ```suggestion
###### Guild Scheduled Event User Add Event Fields
```

| Field                    | Type      | Description                     |
| ------------------------ | --------- | ------------------------------- |

> | guild_scheduled_event_id | snowflake | id of the guild scheduled event |

Reviewer 5: ```suggestion
| guild_scheduled_event_id | snowflake | ID of the guild scheduled event |
```

> | user_id                  | snowflake | id of the user                  |

Reviewer 5: ```suggestion
| user_id                  | snowflake | ID of the user                  |
```

> | guild_id                 | snowflake | id of the guild                 |

Reviewer 5: ```suggestion
| guild_id                 | snowflake | ID of the guild                 |
```

### Guild Scheduled Event User Remove

Sent when a user has unsubscribed from a guild scheduled event.

###### Guild Scheduled Event User Remove Event Fields

| Field                    | Type      | Description                     |
| ------------------------ | --------- | ------------------------------- |

> | guild_scheduled_event_id | snowflake | id of the guild scheduled event |

Reviewer 5: ```suggestion
| guild_scheduled_event_id | snowflake | ID of the guild scheduled event |
```

> | user_id                  | snowflake | id of the user                  |

Reviewer 5: ```suggestion
| user_id                  | snowflake | ID of the user                  |
```

> | guild_id                 | snowflake | id of the guild                 |

Reviewer 5: ```suggestion
| guild_id                 | snowflake | ID of the guild                 |
```
